sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/InternalColumnarRddConverter.scala

```diff
@@ -280,90 +437,125 @@ private object GpuExternalRowToColumnConverter {
-  //        val data = row.getStruct(column, childConverters.length)
-  //        for (i <- 0 until childConverters.length) {
-  //          childConverters(i).append(data, i, cv.getChild(i))
-  //        }
-  //      }
-  //    }
-  //  }
-  //
-  //  private case class DecimalConverter(dt: DecimalType) extends TypeConverter {
-  //    override def append(
-  //        row: SpecializedGetters,
-  //        column: Int,
-  //        builder: ai.rapids.cudf.HostColumnVector.Builder): Unit = {
-  //      if (row.isNullAt(column)) {
-  //        builder.appendNull()
-  //      } else {
-  //        val d = row.getDecimal(column, dt.precision, dt.scale)
-  //        if (dt.precision <= Decimal.MAX_INT_DIGITS) {
-  //          cv.appendInt(d.toUnscaledLong.toInt)
-  //        } else if (dt.precision <= Decimal.MAX_LONG_DIGITS) {
-  //          cv.appendLong(d.toUnscaledLong)
-  //        } else {
-  //          val integer = d.toJavaBigDecimal.unscaledValue
-  //          val bytes = integer.toByteArray
-  //          cv.appendByteArray(bytes, 0, bytes.length)
-  //        }
-  //      }
-  //    }
-  //  }
-  //
-  //  private case class MapConverter(keyConverter: TypeConverter, valueConverter: TypeConverter)
-  //    extends TypeConverter {
-  //    override def append(
-  //        row: SpecializedGetters,
-  //        column: Int,
-  //        builder: ai.rapids.cudf.HostColumnVector.Builder): Unit = {
-  //      if (row.isNullAt(column)) {
-  //        builder.appendNull()
-  //      } else {
-  //        val m = row.getMap(column)
-  //        val keys = cv.getChild(0)
-  //        val values = cv.getChild(1)
-  //        val numElements = m.numElements()
-  //        cv.appendArray(numElements)
-  //
-  //        val srcKeys = m.keyArray()
-  //        val srcValues = m.valueArray()
-  //
-  //        for (i <- 0 until numElements) {
-  //          keyConverter.append(srcKeys, i, keys)
-  //          valueConverter.append(srcValues, i, values)
-  //        }
-  //      }
-  //    }
-  //  }
+
+  private[this] def arrayConvert(
+    childConverter: TypeConverter,
+    row: Row,
+    column: Int,
+    builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder) : Double = {
+    var ret = 0.0
+    val values = row.getSeq(column)
+    val numElements = values.size
+    val child = builder.getChild(0)
+    for (i <- 0 until numElements) {
+      ret += childConverter.append(Row(values: _*), i, child)
+    }
+    builder.endList()
+    ret + OFFSET
+  }
+
+  private case class ArrayConverter(childConverter: TypeConverter)
+    extends TypeConverter {
+    override def append(row: Row,
+      column: Int, builder: ai.rapids.cudf.HostColumnVector.ColumnBuilder): Double = {
+      if (row.isNullAt(column)) {
+        builder.appendNull()
+        VALIDITY_N_OFFSET
+      } else {
+        arrayConvert(childConverter, row, column, builder) + VALIDITY
+      }
+    }
+
+    override def getNullSize: Double = OFFSET + VALIDITY
```

Comment:
Nit: Can use `VALIDITY_N_OFFSET` here, also applies to the instance a few lines below.